### DISCUSSION TITLE
Home Screen customization

### DISCUSSION THREAD

**Discussion by frankwoconnor** [jcorporation/myMPD#1379] [Q&A] [answered]
I'm trying to programmatically modify the home screen. It looks like the display for the home screen is in part driven by this file:
`/var/lib/mympd/state/home_list`
 
 I have manually (via the GUI) created two icons which open up a filtered browser window.

`{"type":"icon","name":"Jazz","ligature":"preview","bgcolor":"#28a745","color":"#f8f9fa","image":"","cmd":"appGoto","options":["Browse","Database","AlbumList","0","100","Genre","{\"tag\":\"AlbumArtist\",\"desc\":false}","Album","((Genre == 'jazz'))"]}
{"type":"icon","name":"melkor","ligature":"preview","bgcolor":"#28a745","color":"#f8f9fa","image":"","cmd":"appGoto","options":["Browse","Database","AlbumList","0","100","Genre","{\"tag\":\"AlbumArtist\",\"desc\":false}","Album","((Genre contains 'blues'))"]}`

I have written a script to pull out the unique genres from MPD and then add these genres as filters to the home screen. However, any manual edits to the home_list file are not reflected in UI, and when I restart myMPD the home_file is reset back to reflect the last changes made from the UI. 

Anyone have any hints/suggestion for me? I'm assuming these UI change are persisted in a DB somewhere?

Thanks

**Comment by jcorporation** [comment, marked as answer]
You should not edit the `/var/lib/mympd/state/home_list` if myMPD is running. myMPD loads the list on startup, keeps it in the memory and writes the list on shutdown. Stop myMPD change the list and start myMPD again.

If you want to modify this list when myMPD is running you should use the API.

**Reply by frankwoconnor** [reply]
Thanks for the quick reply! That worked perfectly.
Just in case It's useful to anyone else, here is the python script that I made.
It connects to MPD, gets a list of unique genres and then creates an icon on the home screen for each genre. 


```
import json
from mpd import MPDClient

# Connect to MPD server
client = MPDClient()
client.connect("localhost", 6600)

# Query MPD for unique genres
unique_genres = client.list("genre")

# Disconnect from MPD server
client.close()
client.disconnect()

# Path to home_list file
home_list_path = '/var/lib/mympd/state/home_list'

# Read the existing home_list file
try:
    with open(home_list_path, 'r') as file:
        home_list = json.loads(f'[{",".join(file.readlines())}]')
except FileNotFoundError:
    home_list = []

# Function to generate a home icon entry
def create_home_icon(genre):
    return {
        "name": genre['genre'],
        "ligature": "preview",
        "bgcolor": "#28a745",
        "color": "#f8f9fa",
        "image": "",
        "cmd": "appGoto",
        "options": [
            "Browse",
            "Database",
            "AlbumList",
            "0",
            "100",
            "Genre",
            "{\"tag\":\"AlbumArtist\",\"desc\":false}",
            "Album",
            f"((Genre contains '{genre['genre']}'))"
        ]
    }

# Add new icons for each genre
for genre in unique_genres:
    home_list.append(create_home_icon(genre))

# Write the updated home_list back to the file
with open(home_list_path, 'w') as file:
    for entry in home_list:
        file.write(json.dumps(entry) + "\n")

print(f"Updated home_list with new icons for each genre.")
```
